concord_cli/kill_task.py

```diff
@@ -1,49 +1,184 @@
 #!/usr/bin/env python
+import sys
 import thrift
 import json
 import logging
-from optparse import OptionParser
+import argparse
 from concord_cli.utils import *
+from terminaltables import AsciiTable
 
 logging.basicConfig()
 logger = logging.getLogger('cmd.kill_task')
 logger.setLevel(logging.INFO)
 
 def generate_options():
-    usage = "usage: %prog [options] arg"
-    parser = OptionParser(usage)
-    parser.add_option("--zookeeper-hosts", dest="zookeeper",
-                      action="store", help="i.e: 1.2.3.4:2181,2.3.4.5:2181")
-    parser.add_option("--task-id", dest="task_id", action="store")
-    parser.add_option("--zookeeper-path", action="store", dest="zk_path")
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-z"
+                        ,"--zookeeper-path"
+                        ,help="Path of concord topology on zk"
+                        ,default="/bolt"
+                        ,action="store")
+    parser.add_argument("-o"
+                        ,"--zookeeper-hosts"
+                        ,help="i.e: 1.2.3.4:2181,2.3.4.5:2181"
+                        ,default="localhost:2181"
+                        ,action="store")
+    parser.add_argument("-t"
+                        ,"--task_id"
+                        ,help="The id of task to kill"
+                        ,action="store")
+    parser.add_argument("-a"
+                        ,"--all"
+                        ,help="Kill all computations"
+                        ,action="store_true")
     return parser
 
-def validate_options(options, parser):
-    if not options.task_id: parser.error("please specify --task-id")
-    if not options.zk_path: parser.error("please specify --zookeeper-path")
-    if not options.zookeeper: parser.error("please specify --zookeeper-hosts")
-
-
-def kill(options):
+def kill(zookeeper, zk_path, *task_ids):
+    if len(task_ids) == 0:
+        logger.info('Kill received an empty list of task_ids')
+        return
     logger.info("Getting master ip from zookeeper")
-    ip = get_zookeeper_master_ip(options.zookeeper, options.zk_path)
+    ip = get_zookeeper_master_ip(zookeeper, zk_path)
     logger.info("Found leader at: %s" % ip)
     (addr, port) = ip.split(":")
     logger.info("Initiating connection to scheduler")
     cli = get_sched_service_client(addr,int(port))
     logger.info("Sending request to scheduler")
-    try: cli.killTask(options.task_id)
+    try:
+        for task_id in task_ids:
+            logger.info('Sending request to kill task: %s', task_id)
+            cli.killTask(task_id)
     except thrift.Thrift.TApplicationException as e:
-        logger.error("Error killing task:%s" % options.task_id)
+        logger.error("Error killing task: %s" % task_id)
```

Comment:
maybe also log the task ids that weren't yet killed? not incredibly important, but useful